### PR TITLE
feat: Impl quick send async

### PR DESCRIPTION
**Pull request by driftluo**
1. impl quick send async
2. use substream as a compound word

dependence #247

### CODE REVIEW

**Comment by doitian** [comment]
> use substream as a compound word

Any reason about this change?

File: src/channel/sink_impl.rs [hunk 1]
```diff
@@ -74,3 +75,73 @@ impl<T> Sink<T> for &UnboundedSender<T> {
...
+    where
+        Self: Unpin,
+    {
+        QuickSend::new(self, item)
+    }
+}
+
+impl<T> QuickSinkExt<T> for Sender<T> {
+    fn quick_start_send(&mut self, msg: T) -> Result<(), Self::Error> {
+        (*self).start_quick_send(msg)
```
Comment (doitian) [inline src/channel/sink_impl.rs:92]: What's the difference between `quick_start_send` and `start_quck_send`? Can they use the same name to be consistent?

**Comment by driftluo** [comment]
> > use substream as a compound word
> 
> Any reason about this change?

Uniform variable style

File: src/channel/sink_impl.rs [hunk 2]
```diff
@@ -74,3 +75,73 @@ impl<T> Sink<T> for &UnboundedSender<T> {
         Poll::Ready(Ok(()))
     }
 }
+
+pub trait QuickSinkExt<Item>: Sink<Item> {
```
Comment (doitian) [inline src/channel/sink_impl.rs:79]: It is a public trait and deserves some documents about the two interfaces.

From the name, I guess that `quick_start_send` is called to start the sink, but it is wired that it also requires the item.

From the code, I see that `quick_send` will call `quick_start_send`. It seems that the names are misleading me.

File: src/channel/sink_impl.rs [hunk 3]
```diff
@@ -74,3 +75,73 @@ impl<T> Sink<T> for &UnboundedSender<T> {
...
+                Poll::Pending => {
+                    this.item = Some(item);
+                    return Poll::Pending;
+                }
+            }
+        }
+
+        // we're done sending the item, but want to block on flushing the
+        // sink
+        ready!(Pin::new(&mut this.sink).poll_flush(cx))?;
```
Comment (doitian) [inline src/channel/sink_impl.rs:143]: Just curious, will it block when there's nothing to flush?

File: src/channel/sink_impl.rs (hunk 2 above)
Comment (doitian) [reply src/channel/sink_impl.rs:79]: I got the reason. It follows the convention used in tokio Sink.

I think name `start_quick_send` is better than `quick_start_send`.

Comment (doitian) [reply src/channel/sink_impl.rs:79]: A quick way to document them: Copy document from tokio and refer to send/start_send.

File: src/channel/sink_impl.rs (hunk 3 above)
Comment (driftluo) [reply src/channel/sink_impl.rs:143]: yes

**Comment by doitian** [comment]
Waiting for rebase and the discussion on the naming and documentation.

**Pull request merged**